ui/pages/providers/[id].js

```diff
@@ -1,14 +1,130 @@
-import { useEffect, useRef, useState } from 'react'
+import { Fragment, useEffect, useRef, useState } from 'react'
+import { Transition, Dialog } from '@headlessui/react'
 import { useRouter } from 'next/router'
+import copy from 'copy-to-clipboard'
 import Head from 'next/head'
 import Link from 'next/link'
 import useSWR, { useSWRConfig } from 'swr'
+import { DuplicateIcon, CheckIcon } from '@heroicons/react/outline'
 import dayjs from 'dayjs'
 
 import Dashboard from '../../components/layouts/dashboard'
 import RemoveButton from '../../components/remove-button'
 import Notification from '../../components/notification'
 
+function SCIMKeyDialog(props) {
+  const [scimAccessKey, setSCIMAccessKey] = useState('')
+  const [error, setError] = useState('')
+  const [keyCopied, setKeyCopied] = useState(false)
+
+  async function onSubmit(e) {
+    e.preventDefault()
+
```

Comment:
do we want to reset the error state here too?

Reply:
good catch